TFEngine.py

```diff
@@ -1344,48 +1345,84 @@ def _do_save(self):
         return False
     return True
 
-  def eval_model(self, output_file=None, output_per_seq_file=None):
+  def eval_model(self, output_file=None, output_per_seq_file=None, loss_name=None, output_per_seq_format="seq_len;score;pos_score"):
     """
     Eval the current model on the eval datasets (dev + eval, whatever is set).
     See also :func:`self.search` for performing beam search.
 
     :param str|None output_file: if given, will save the results to this file (total err/score for each dataset)
     :param str|None output_per_seq_file: if given, will save the err/score for each sequence
+    :param [str]|None loss_name=None: Needed if output_file is given.
+    :param str|None output_per_seq_format="seq_len;score;pos_score":
+                                            which properies of `loss_name` should be written to `output_per_seq_file`.
+                                            allowed_outputs = {"seq_tags", "seq_len", "score", "error", "pos_score", "pos_error"}
+                                            requested outputs must be separated by a colon ";", e.g. "seq_len;score;error"
```

Comment:
I would remove some indent spaces here. All the contiguous comments starting in the next line can be just 2 spaces indented from the first line. Like this:
```
:param str foo: comment.
  comment continues here.
  and here some more.
```

Reply:
Also, I really would not introduce custom syntax. This can become ugly. Just require a `list` or `tuple`.